server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java

```diff
@@ -81,18 +88,108 @@ public String source() {
      * @param source                  the source of the retention lease
      */
     public RetentionLease(final String id, final long retainingSequenceNumber, final long timestamp, final String source) {
+        Objects.requireNonNull(id);
+        if (id.contains(":") || id.contains(";") || id.contains(",")) {
```

Comment:
I think both `id` and `source` should be non-empty; otherwise, we fail to decode it?

Reply:
I pushed 444f483f0b2caa9f4303a13a6054257255035539.